[coregano] iP

coregano opened the pull request.

coregano committed:
level-0 complete

coregano committed:
completed Level-1

coregano committed:
level-2 intermediate

coregano committed:
Level-2 complete, removed class file Listing for cleaner coding

coregano committed:
Level-2 complete, mistake made during staging

coregano committed:
Level-3 complete, added task class

coregano committed:
Added a few static final variables and comments for better code readability

coregano committed:
Added a few string constants and improved readability

coregano committed:
Level-4 complete, added "help" function as well

keanecxz commented on src/main/java/Duke.java:
```diff
@@ -1,103 +1,222 @@
-                LINE_PARTITION);
+    private static String[] splitUserInput(String rawUserInput) {
+        final String[] split = rawUserInput.trim().split("\\s+", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""}; // else case: no parameters
```
Maybe you could move the comment along the same indentation so as to avoid breaking the logical structure of the program.

keanecxz commented on src/main/java/Duke.java:
```diff
@@ -1,103 +1,222 @@
+        System.out.print(String.format("\t  Please use a number between 1 and %d for \"mark\" and \"unmark\" commands\n", listLength) +
+                LINE_PARTITION);
+    }
+    private static void noMarkArgs() {
```
The name of this method may be unclear as it does not indicate an action. Maybe add a verb such as print to become printNoMarkArgs. I believe that would make the purpose of the function clearer.

keanecxz commented on src/main/java/Duke.java:
```diff
@@ -1,103 +1,222 @@
+    private static final Scanner in = new Scanner(System.in);
+
+    // max no. of tasks in taskList
+    private static final int MAX_TASKS = 100;
```
Wonderful job for keeping consistency of capital cases for all your constant values.

keanecxz commented on src/main/java/Duke.java:
```diff
@@ -1,103 +1,222 @@
+            System.out.print(LINE_PARTITION +
+                    "Invalid input for \"mark\" command. Please try again." +
+                    LINE_PARTITION);
         }
```
Each switch case has clear purpose, well done!

keanecxz commented on src/main/java/Duke.java:
```diff
@@ -1,103 +1,222 @@
+        listLength++;
+        printTaskAdd();
+    }
+    private static void cmdFormatError() {
```
Maybe a verb can be used to increase clarity of function. I suggest printCmdError.

keanecxz reviewed (commented):
Very clear and readable code! Made a few suggestion on coding standard

coregano committed:
Bug fix for help message and switch case.

coregano committed:
Level-5 complete

coregano committed:
Merge branch 'branch-Level-5'

coregano committed:
Merged branch-Level-5 back into master.
Sorted classes into packages.
Added and edited code comments.

siyuancheng178 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,260 @@
+		final String userInput = in.nextLine();
+		final String[] cmdTypeAndParams = splitUserInput(userInput);
+		final String cmd = cmdTypeAndParams[0];
+		final String args = cmdTypeAndParams[1];
```
Maybe these variables are not required to be `final`? Or variables with `final` should be CAPITALISED.

siyuancheng178 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,260 @@
+        "| | | | | | | |/ / _ \\\n" +
+        "| |_| | |_| |   <  __/\n" +
+        "|____/ \\__,_|_|\\_\\___|\n";
+System.out.println("Hello from\n" + logo);
```
Please delete these commented codes

coregano committed:
level-6 complete on branch-level-6

coregano committed:
Level-7 stage 1 complete

coregano committed:
Level-7 intermediate, saving done but not tested

coregano committed:
Level-7 complete, save tested

coregano committed:
Merge branch 'branch-level-6'

coregano committed:
Merge branch 'branch-level-7' after 'branch-level-6' was merged

# Conflicts:
#	src/main/java/Duke.java

coregano committed:
Bug fixes, text formatting complete.

coregano committed:
Ver 1.1
Specified java library imports, ready for jar file testing.

coregano committed:
Added taskList save file to git.ignore list

coregano committed:
Added:
Command package
Cmd classes
Parse package and classes
Ui pakcage and classes

Other:
Rearranged classes and packages
Started on Cmd classes

coregano committed:
Created commands and tasklist classes and subclasses

coregano committed:
Edit code for all subclasses of Cmd
Fix bug of DeleteCmd returning IndexOutOfBounds exception

coregano committed:
Add find function
Edit code on Parser
Add Javadocs
Remove bugs regarding delete and numberformatexception

coregano committed:
Update readme
